Xplat/src/main/java/vazkii/botania/common/block/flower/functional/LooniumBlockEntity.java

```diff
@@ -60,7 +62,49 @@ public class LooniumBlockEntity extends FunctionalFlowerBlockEntity {
 			Zombie.class
 	);
 
-	private ResourceLocation lootTable = new ResourceLocation("minecraft", "chests/simple_dungeon");
+	private record StructureLoot(ResourceLocation loc, int weight) {
+		public StructureLoot(ResourceLocation loc) {
+			this(loc, 1);
+		}
+	}
+
+	//TODO make this pull from a JSON file in a datapack instead of being hardcoded
+	private static final Map<ResourceLocation, StructureLoot> lootTables = Map.of(
+			new ResourceLocation("minecraft:ancient_city"), new StructureLoot(new ResourceLocation("botania:loonium/ancient_city")),
+			new ResourceLocation("minecraft:bastion"), new StructureLoot(new ResourceLocation("botania:loonium/bastion")),
+			new ResourceLocation("minecraft:end_city"), new StructureLoot(new ResourceLocation("botania:loonium/end_city")),
+			new ResourceLocation("minecraft:fortress"), new StructureLoot(new ResourceLocation("minecraft:nether_bridge")),
+			new ResourceLocation("minecraft:stronghold"), new StructureLoot(new ResourceLocation("botania:loonium/stronghold")),
+			new ResourceLocation("minecraft:mansion"), new StructureLoot(new ResourceLocation("minecraft:woodland_mansion"))
+	);
+	private static final ResourceLocation defaultLootTable = new ResourceLocation("minecraft", "chests/simple_dungeon");
+
+	private LootTable getLootTable(ServerLevel level) {
+		List<StructureLoot> validLoot = new ArrayList<StructureLoot>();
+		int totalWeight = 0;
+		for (var entry : lootTables.entrySet()) {
+			ResourceLocation key = entry.getKey();
+			StructureLoot value = entry.getValue();
+			Structure structure = StructureHelper.getStructure(level, key);
+			if (StructureHelper.isInStructureBounds(level, getEffectivePos(), structure)) { //could be getBlockPos
+				validLoot.add(value);
+				totalWeight += value.weight();
+			}
+		}
+		if (totalWeight > 0) {
+			int roll = level.random.nextInt(totalWeight);
+			int i = 0;
+			while (true) {
+				roll -= validLoot.get(i).weight();
+				if (roll < 0) {
+					break;
+				}
+			}
```

Comment:
No need to roll your own weighted select, vanilla has utilities for it. Look around for stuff called `WeightedRandom` or similar